sqlpage/templates/shell.handlebars

```diff
@@ -49,76 +49,79 @@
-            </div>
-        </nav>
+		<header id="sqlpage_header">
+			<nav class="navbar navbar-expand-md navbar-light">
+				<div class="container-fluid">
+					<a class="navbar-brand flex-grow-1 overflow-hidden" href="{{#if link}}{{link}}{{else}}/{{/if}}">
+						{{#if image}}
+							<img src="{{image}}" alt="{{title}}" width="32" height="32"
+								class="navbar-brand-image">
+						{{/if}}
+						{{#if icon}}
+							{{~icon_img icon~}}
+						{{/if}}
+						<h1 class="mb-0 w-0 fs-2">{{title}}</h1>
+					</a>
+					<button class="navbar-toggler" type="button" data-bs-toggle="collapse"
+						data-bs-target="#navbar-menu" aria-controls="navbar-menu" aria-expanded="false"
+						aria-label="Toggle navigation">
+						<span class="navbar-toggler-icon"></span>
+					</button>
+					<div class="collapse navbar-collapse" id="navbar-menu">
+						<ul class="navbar-nav ms-auto">
+							{{#each (to_array menu_item)}}
+								{{#if (eq (typeof this) 'object')}}
+									<li class="nav-item{{#if this.submenu}} dropdown{{/if}}">
+									<a class="nav-link {{#if this.submenu}}dropdown-toggle{{/if}}" href="{{#if this.link}}{{this.link}}{{else}}#{{/if}}"
+										{{#if this.submenu}}data-bs-toggle="dropdown" data-bs-auto-close="outside"{{/if}}
+										role="button"
+									>
+										{{this.title}}
+									</a>
+									{{#if this.submenu}}
+										<div class="dropdown-menu dropdown-menu-end" data-bs-popper="static">
+											{{#each this.submenu}}
+												<a class="dropdown-item" href="{{this.link}}">
+													{{this.title}}
+												</a>
+											{{/each}}
+										</div>
+									{{/if}}
+									</li>
+								{{else}}
+									<li class="nav-item">
+										<a class="nav-link text-capitalize" href="{{this}}.sql">{{this}}</a>
+									</li>
+								{{/if}}
+							{{/each}}
+						</ul>
+						{{#if search_target}}
+							<form class="d-flex" role="search" action="{{search_target}}">
+								<input class="form-control me-2" type="search" placeholder="Search" aria-label="Search" name="search" value="{{search_value}}">
+								<button class="btn btn-outline-success" type="submit">Search</button>
+							</form>
+						{{/if}}
+					</div>
+				</div>
+			</nav>
+		</header>
     {{/if}}
-    <main class="page-wrapper container-xl pt-3 px-md-5 px-sm-3">
+    <main class="page-wrapper container-xl pt-3 px-md-5 px-sm-3" id="sqlpage_main_wrapper">
         {{~#each_row~}}{{~/each_row~}}
     </main>
 </div>
-<div class="w-100 text-center fs-6 my-2 text-secondary">
-    {{#if footer}}
-        {{{markdown footer}}}
-    {{else}}
-        <!-- You can change this footer using the 'footer' parameter of the 'shell' component -->
-        Built with <a class="text-reset" href="https://sql.ophir.dev" title="SQLPage v{{buildinfo 'CARGO_PKG_VERSION'}}">SQLPage</a>
-    {{/if}}
-</div>
+<footer class="w-100 text-center fs-6 my-2 text-secondary" id="sqlpage_footer">
+	<div class="sqlpage_footer_content">
```

Comment:
Same as before, I'm not a huge fan of adding nesting if there isn't a good reason. Do we need footer + div, or can we go with just a footer ?

Reply:
Regarding these two 👆: mostly to include semantics (`header` and `footer`) but also to provide _some_ structure for those who want to shape the layout a bit with CSS, if desired.

I'm not a fan of meaningless divs either, but I do like _some_ internal structure (as of now there is none). Having at least one internal div gives us a lot more flexibility for CSS layout, backgrounds, margin/padding, etc, without making a noticeable different if we don't use CSS (since the blocks collapse if not styled).

What do you think?

Reply:
Do you have an example of something you wanted to do that is harder without the internal div ?